client-c/main.cpp

```diff
@@ -155,12 +153,29 @@ void runKServer(httplib::Server *svr) {
   scheduleinj->h = injHeaderSchedule;
   scheduleinj->data = (block*)schedule;
 
-  int sock = init(port, address);
-  zinj *sockinj = (zinj *)koreAlloc(sizeof(zinj));
-  sockinj->h = injHeaderInt;
-  mpz_t sock_z;
-  mpz_init_set_si(sock_z, sock);
-  sockinj->data = move_int(sock_z);
+  int input[2], output[2];
+  if (pipe(input)) {
+    perror("pipe");
```

Comment:
can we have more specific errors? maybe even just say "input pipe" instead of just "pipe"